src/App.tsx

```diff
@@ -1,27 +1,93 @@
-import React from 'react';
+import React, {
+  useCallback, useEffect, useRef, useState,
+} from 'react';
+import { getFiveFirst, getAll, getRed } from './api/goods';
 import './App.scss';
 import { GoodsList } from './GoodsList';
+import { Good } from './types/Good';
 
 // import { getAll, get5First, getRed } from './api/goods';
 // or
 // import * as goodsAPI from './api/goods';
 
-export const App: React.FC = () => (
-  <div className="App">
-    <h1>Dynamic list of Goods</h1>
+export const App: React.FC = () => {
+  const [goods, setGoods] = useState<Good[]>([]);
+  const mountedRef = useRef(false);
 
-    <button type="button" data-cy="all-button">
-      Load all goods
-    </button>
+  useEffect(() => {
+    let mounted = true;
 
-    <button type="button" data-cy="first-five-button">
-      Load 5 first goods
-    </button>
+    mountedRef.current = true;
 
-    <button type="button" data-cy="red-button">
-      Load red goods
-    </button>
+    const load = async (): Promise<void> => {
+      const newGoods = await getAll();
 
-    <GoodsList goods={[]} />
-  </div>
-);
+      if (mounted) {
+        setGoods(newGoods);
+      }
+    };
+
+    load();
+
+    return () => {
+      mounted = false;
+      mountedRef.current = false;
+    };
+  }, []);
+
+  const handleButtonAllClicked = useCallback(async (): Promise<void> => {
```

Comment:
useCallback is useless in those cases 
it's a complex topic so you can read more here if you want to dive deeper
https://www.freecodecamp.org/news/memoization-in-javascript-and-react/